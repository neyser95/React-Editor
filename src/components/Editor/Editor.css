.EditorLayout{
  display: flex;
  border: solid 1px rgba(0, 0, 0, 0.7);
  width: 100%;
  height: 100vh;
  box-sizing: border-box;
  margin: 10px 0;
}
.Editor{
  width: 100%;
  padding: 10px;
  box-sizing: border-box;
}
.Editor__Split{
  width: 50%;
  padding: 10px;
  box-sizing: border-box;
}

.MarkupEditor{
  border-right: solid 1px rgba(0, 0, 0, 0.7);
  width: 50%;
  padding: 10px;
  box-sizing: border-box;
}

/* * Will only print content within the editor */
@media print {
  body * {
    visibility: hidden;
  }

  .Editor, .Editor * {
    visibility: visible;
  }

  .Editor {
    position: absolute;
    border: none;
    left: 0;
    top: 0;
  }
}