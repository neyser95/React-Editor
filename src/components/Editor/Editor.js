import React from 'react';
import classNames from './Editor.css';

const editor = (props) => {
  return (
    // * Will create a div whose content will be editable.
    <div
      className={classNames.Editor}
      onInput={(e) => props.handleEditorChange(e)} 
      contentEditable>
    </div>
  );
}

export default editor;