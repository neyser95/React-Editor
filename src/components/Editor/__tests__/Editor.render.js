import React from 'react';
import renderer from 'react-test-renderer';
import Editor from '../Editor';


//* Tests to make sure that comopnent does not change unexpectdly.
test('Editor will render with regular editor', () => {
  const editor = renderer.create(
    <Editor
      markdownEnabled={false}
    />
  );

  let editorComponent = editor.toJSON();
  expect(editorComponent).toMatchSnapshot();
});

test('Editor will render with markup editor and editor display', () => {
  const editor = renderer.create(
    <Editor
      markdownEnabled={true}
      parsedMarkdown={'<h1>Hello</h1>'}
    />
  );

  let editorComponent = editor.toJSON();
  expect(editorComponent).toMatchSnapshot();
});