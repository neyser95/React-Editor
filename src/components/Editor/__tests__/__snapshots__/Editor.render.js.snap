// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor will render with markup editor and editor display 1`] = `
<div>
  <div
    contentEditable={true}
    onInput={[Function]}
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>Hello</h1>",
      }
    }
    onInput={[Function]}
  />
</div>
`;

exports[`Editor will render with regular editor 1`] = `
<div>
  <div
    contentEditable={true}
    onInput={[Function]}
  />
</div>
`;
